modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java

```diff
@@ -331,6 +344,170 @@ public ModelsMap postProcessModels(ModelsMap objs) {
+        final String kSelfOnlyProps = "x-self-only-props";
+        final String kHasSelfOnlyProps = "x-has-self-only-props";
+        final String kAncestorOnlyProps = "x-ancestor-only-props";
+        final String kHasAncestorOnlyProps = "x-has-ancestor-only-props";
+        final String kSelfAndAncestorOnlyProps = "x-self-and-ancestor-only-props";
+        final String kHasSelfAndAncestorOnlyProps = "x-has-self-and-ancestor-only-props";
+
+        Set<String> allPureClasses = new HashSet<>();
+        //set isChild,isParent,isPure
+        for (java.util.Map.Entry<String,CodegenModel> cmEntry : allModels.entrySet()) {
+            String key = cmEntry.getKey();
+            CodegenModel cm = cmEntry.getValue();
+            //get all ancestors
+            Set<String> allAncestors = allAncestorsForAllModels.get(key);
+            
+            //a class is a parent when it's an ancestor to another class
+            boolean isParent = allAncestorsForAllModelsFlat.contains(key);
+            //a class is a child when it has any ancestor
+            boolean isChild = !allAncestors.isEmpty();
+            //a class is pure when it's not a child, and has no oneOf nor anyOf
+            boolean isPure = !isChild && (cm.oneOf == null || cm.oneOf.isEmpty()) && (cm.anyOf == null || cm.anyOf.isEmpty()) ;
+            
+            cm.vendorExtensions.put(kIsChild, isChild);
+            cm.vendorExtensions.put(kIsParent, isParent);
+            cm.vendorExtensions.put(kIsPure, isPure);
+            //when pure:
+            //vars = allVars = selfOnlyProperties = kSelfAndAncestorOnlyProps
+            //ancestorOnlyProps = empty
+            if (isPure) {
+                cm.vendorExtensions.put(kSelfOnlyProps, cm.getVars());
+                cm.vendorExtensions.put(kHasSelfOnlyProps, !cm.getVars().isEmpty());
+                cm.vendorExtensions.put(kAncestorOnlyProps, new ArrayList<CodegenProperty>());
+                cm.vendorExtensions.put(kHasAncestorOnlyProps, false);
+                cm.vendorExtensions.put(kSelfAndAncestorOnlyProps,  cm.getVars());
+                cm.vendorExtensions.put(kHasSelfAndAncestorOnlyProps,  !cm.getVars().isEmpty());
+
+                allPureClasses.add(key);
+            }
+        }
+
+        //handle impure models
+        for (java.util.Map.Entry<String,CodegenModel> cmEntry : allModels.entrySet()) {
+            String key = cmEntry.getKey();
+            CodegenModel cm = cmEntry.getValue();
+            if (allPureClasses.contains(key)) {
+                continue;
+            }
+            //get all ancestors
+            Set<String> allAncestors = allAncestorsForAllModels.get(key);
+         
+            //get direct parents
+            // Set<String> directParentNames = cm.allOf == null ? new HashSet<>() : cm.allOf;
+            Set<String> compositeProperties = new HashSet<>();
+            
+            Set<String> compositeModelNames = new HashSet<String>() {
```

Comment:
this pattern of adding elements to collections is really slow and unknown to many programmers, can we have standard `compositeModelNames.add...` please?